src/main/java/com/amazon/opendistroforelasticsearch/ad/model/AnomalyDetector.java

```diff
@@ -179,6 +180,66 @@ public AnomalyDetector(
         );
     }
 
+    public AnomalyDetector(
+        String detectorId,
+        Long version,
+        String name,
+        String description,
+        String timeField,
+        List<String> indices,
+        List<Feature> features,
+        QueryBuilder filterQuery,
+        TimeConfiguration detectionInterval,
+        TimeConfiguration windowDelay,
+        Integer shingleSize,
+        Map<String, Object> uiMetadata,
+        Integer schemaVersion,
+        Instant lastUpdateTime,
+        List<String> categoryFields,
+        User user,
+        String detectorType
+    ) {
+        if (Strings.isBlank(name)) {
+            throw new IllegalArgumentException("Detector name should be set");
+        }
+        if (timeField == null) {
+            throw new IllegalArgumentException("Time field should be set");
+        }
+        if (indices == null || indices.isEmpty()) {
+            throw new IllegalArgumentException("Indices should be set");
+        }
+        if (detectionInterval == null) {
+            throw new IllegalArgumentException("Detection interval should be set");
+        }
+        if (shingleSize != null && shingleSize < 1) {
+            throw new IllegalArgumentException("Shingle size must be a positive integer");
+        }
+        if (categoryFields != null && categoryFields.size() > CATEGORY_FIELD_LIMIT) {
+            throw new IllegalArgumentException(CommonErrorMessages.CATEGORICAL_FIELD_NUMBER_SURPASSED + CATEGORY_FIELD_LIMIT);
```

Comment:
minor - if we have some constant `CommonErrorMessages` used for the category field here, should we do the same for the above fields, rather than hardcoded strings? I'm thinking this could help when adding new constructors for the detector, and to make sure we're consistent & grammatically correct since some of these may be propagated on the frontend through toasts or callouts

Reply:
Good point, there is some overlap between this PR and Yizhe's validation PR  https://github.com/opensearch-project/anomaly-detection/pull/20/files#diff-ab4bf6eb04e845b21b1182e30afb84c73b282bae5cc26dbcfef2e846211bcdecR213 . I will add these error message to `CommonErrorMessages`, @yizheliu-amazon , you can make similar change in your PR or merge after this PR merged, ,

Reply:
sure. I can merge it after this PR merged.